mesh2hrtf/Mesh2Input/mesh2input.py

```diff
@@ -1160,6 +1175,48 @@ def _render_pictures(filepath1, unitFactor):
         nameRender = ("%d_deg_azimuth" % azim[ii])
         bpy.data.images['Render Result'].save_render(
             os.path.join(dirRender, "%s.png" % nameRender))
+        
+
+#Added by Sanket
+def _write_checkall_txt(filepath1):
+    """check all before export"""
+    #check for geometry is solid
+    bpy.ops.mesh.print3d_check_solid()
+    info = report.info()
+    non_manifold = info[0][0]
+    non_manifold_num = int(re.search(r'\d+', non_manifold).group())
+    bad_contiguous = info[1][0]
+    bad_contiguous_num = int(re.search(r'\d+', bad_contiguous).group())
+    #check geometry for self intersections
+    bpy.ops.mesh.print3d_check_intersect()
+    info = report.info()
+    intersect_face = info[0][0]
+    intersect_face_num = int(re.search(r'\d+', intersect_face).group())
+    #check edges are below the sharpness preference
+    bpy.ops.mesh.print3d_check_sharp()
+    info = report.info()
+    sharp_edge = info[0][0]
+    sharp_edge_num = int(re.search(r'\d+', sharp_edge).group())
```

Comment:
Sharp edges might indeed be a problem in the BEM. Wolfgang might have an idea about how critical this is. Can the threshold for reporting a sharp edge be set?